crates/sui-json-rpc/src/read_api.rs

```diff
@@ -160,7 +160,18 @@ impl ReadApiServer for ReadApi {
 
         Ok(SuiTransactionResponse {
             transaction: transaction.into_message().try_into()?,
-            effects: SuiTransactionEffects::try_from(effects, self.state.module_cache.as_ref())?,
```

Comment:
this is also a question for @patrickkuo about what module caches may be needed here and whether this call may come across epochs for different epochs

Reply:
though this always using the latest according to state should be just fine